hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncTableImpl.java

```diff
@@ -329,43 +348,52 @@ private void preCheck() {
     public CompletableFuture<Boolean> thenPut(Put put) {
       validatePut(put, conn.connConf.getMaxKeyValueSize());
       preCheck();
+      final Supplier<Span> supplier = new TableOperationSpanBuilder()
+        .setTableName(tableName)
+        .setOperation(HBaseSemanticAttributes.Operation.CHECK_AND_MUTATE);
       return tracedFuture(
         () -> RawAsyncTableImpl.this.<Boolean> newCaller(row, put.getPriority(), rpcTimeoutNs)
           .action((controller, loc, stub) -> RawAsyncTableImpl.mutate(controller, loc, stub, put,
             (rn, p) -> RequestConverter.buildMutateRequest(rn, row, family, qualifier, op, value,
               null, timeRange, p, HConstants.NO_NONCE, HConstants.NO_NONCE),
             (c, r) -> r.getProcessed()))
           .call(),
-        "AsyncTable.CheckAndMutateBuilder.thenPut", tableName);
+        supplier);
     }
 
     @Override
     public CompletableFuture<Boolean> thenDelete(Delete delete) {
       preCheck();
+      final Supplier<Span> supplier = new TableOperationSpanBuilder()
+        .setTableName(tableName)
+        .setOperation(HBaseSemanticAttributes.Operation.CHECK_AND_MUTATE);
       return tracedFuture(
         () -> RawAsyncTableImpl.this.<Boolean> newCaller(row, delete.getPriority(), rpcTimeoutNs)
           .action((controller, loc, stub) -> RawAsyncTableImpl.mutate(controller, loc, stub, delete,
             (rn, d) -> RequestConverter.buildMutateRequest(rn, row, family, qualifier, op, value,
               null, timeRange, d, HConstants.NO_NONCE, HConstants.NO_NONCE),
             (c, r) -> r.getProcessed()))
           .call(),
-        "AsyncTable.CheckAndMutateBuilder.thenDelete", tableName);
+        supplier);
     }
 
     @Override
-    public CompletableFuture<Boolean> thenMutate(RowMutations mutation) {
+    public CompletableFuture<Boolean> thenMutate(RowMutations mutations) {
       preCheck();
-      validatePutsInRowMutations(mutation, conn.connConf.getMaxKeyValueSize());
+      validatePutsInRowMutations(mutations, conn.connConf.getMaxKeyValueSize());
+      final Supplier<Span> supplier = new TableOperationSpanBuilder()
+        .setTableName(tableName)
+        .setOperation(HBaseSemanticAttributes.Operation.BATCH);
```

Comment:
[nit] I found you have a class instance check for  `RowMutations` , can't we just use `.setOperation(mutations)` here?

Reply:
No -- this is confusing -- we're both wrong. The operation of the `CheckAndMutateBuilder` instance is a `CHECK_AND_PUT`. That operation is a "container" that ships multiple underlying operations. `thenMutate` allows the `CHECH_AND_PUT` to apply the contents of `mutations` ; `thenDelete` sends a `DELETE`, &c.

My patch on HBASE-26473 introduces the span attribute `db.hbase.container_operations` as a general mechanism for handling these operations that wrap other operations. That commit is on the feature branch PR that I posted earlier, https://github.com/apache/hbase/pull/3863/commits/ac1cfbbec75e2a9a8ca1282cce307db0ab72c3eb

Reply:
ack!

Reply:
Since we haven't gotten to that PR yet, I'm asking in the community what they suggest for this type of operation. https://cloud-native.slack.com/archives/C01QZFGMLQ7/p1638564279059600

Reply:
Latest patch corrects the operation type for the checkandmutate cases @taklwu pointed out. thanks for noticing!